lib/model/query/audits.js

```diff
@@ -115,23 +115,32 @@ const _getByEntityId = (fields, options, entityId) => sql`
 SELECT ${fields} FROM audits
   LEFT JOIN actors ON actors.id=audits."actorId"
 
-  LEFT JOIN audits triggering_event ON (audits.details->'eventId')::INTEGER = triggering_event.id 
+  LEFT JOIN entity_defs ON (audits.details->'entityDefId')::INTEGER = entity_defs.id
+  LEFT JOIN entity_def_sources on entity_def_sources.id = entity_defs."sourceId"
+
+  LEFT JOIN audits triggering_event ON entity_def_sources."auditId" = triggering_event.id
   LEFT JOIN actors triggering_event_actor ON triggering_event_actor.id = triggering_event."actorId"
 
+  -- if triggering event has a submissionId defined, look up creation event for that submission
+  -- it has info about the submission and creator we want to show even if the submission is deleted
   LEFT JOIN audits submission_create_event ON (triggering_event.details->'submissionId')::INTEGER = (submission_create_event.details->'submissionId')::INTEGER AND submission_create_event.action = 'submission.create'
   LEFT JOIN actors submission_create_event_actor ON submission_create_event_actor.id = submission_create_event."actorId"
 
+  -- if source submissionDefId is defined:
   LEFT JOIN (
     (
-      SELECT submissions.*, submission_defs."userAgent" FROM submissions
-      JOIN submission_defs ON submissions.id = submission_defs."submissionId" AND root AND submissions."deletedAt" IS NULL
+      SELECT submissions.*, '' as "userAgent"FROM submissions
     ) submissions
-      JOIN forms ON forms.id = submissions."formId" AND forms."deletedAt" IS NULL  
-  ) ON (triggering_event.details->'submissionId')::INTEGER = submissions.id
+    JOIN forms ON forms.id = submissions."formId" AND forms."deletedAt" IS NULL
+    JOIN submission_defs ON submissions.id = submission_defs."submissionId" AND submissions."deletedAt" IS NULL
+  ) on submission_defs.id = entity_def_sources."submissionDefId"
```

Comment:
this will return the submission_def that created the entity, but we have to return latest instanceName